### PR TITLE
Dashcard visualization settings + multiseries cleanup

### PR DESCRIPTION
**Pull request by tlrobinson**
- [x] Adds visualization settings to dash cards
- [x] Improved dashcard multiseries to respect column selections, etc
- [x] Adds a funnel chart type, in preparation for #3275
- [x] Centralized functionality for transforming/delegating visualizations
- [x] Fix timeseries formatting issue due to b58dc459c75bd1628faf5df0559e32ca3d3329d3

### CODE REVIEW

**Comment by tlrobinson** [comment]
@kdoh @mazameli We're going to need icons for scatter, funnel, and progress. Do those exist yet?

**Comment by kdoh** [comment]
@tlrobinson. We have one for funnel already. I'll double check but I think we have one for progress. and I may need to do one for scatterplots.

One other thing we'll need to take into account is the fact that elsewhere in the app we're using the relatively older "illustrated" style of iconography for different chart types (homepage recents sidebar item and the add to dashboard list). We'll want to either replace or update those accordingly.

**Comment by mazameli** [comment]
- If I edit a dashcard (e.g., I changed a line chart’s display to not show the y-axis labels), it seems I can no longer click the card title to go to the card detail
- Scatterplot bubbles and curved line charts should maybe be cropped so that they don’t hide axis labels? Particularly x-axis labels:

screen shot 2016-09-30 at 2 54 40 pm

screen shot 2016-09-30 at 2 56 34 pm

File: frontend/src/metabase/visualizations/Funnel.jsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    static iconName = "funnel";
+
+    static minSize = { width: 3, height: 3 };
+    static noHeader = true;
+
+    static isSensible(cols, rows) {
+        return true;
+    }
+
+    static checkRenderable(cols, rows) {
```
Comment (kdoh) [inline frontend/src/metabase/visualizations/Funnel.jsx]: Looks like this method is empty.

**Comment by camsaul** [comment]
Ok I fixed the merge conflict.

@tlrobinson since we already had a migration 42 I changed yours to 45. You'll probably need to manually fix change the migration row locally or just drop the new visualization_settings column and run the migrations again.

File: src/metabase/api/dashboard.clj [hunk 2]
```diff
@@ -89,6 +89,7 @@
   (read-check Card cardId)
   (let [defaults       {:dashboard_id id
                         :card_id      cardId
+                        :visualization_settings {}
```
Comment (camsaul) [inline src/metabase/api/dashboard.clj]: ack 😒

**Comment by camsaul** [comment]
@salsakran k I'll fixed I'll go ahead and merge

**Pull request merged**